### PR TITLE
Load plugin tasks first

### PR DESCRIPTION
**Pull request by repher**
load plugin tasks first to prefer custom tasks with the same name in
case of an overwritten plugin task

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
 	],
 	"require": {
 		"cakephp/cakephp": "~3.0",
-		"php": ">=5.4.16"
+		"php": ">=5.4.16",
+		"phpunit/phpunit": "^5.1@dev"
```
Comment (dereuromark) [inline composer.json]: why that? I prefer using the phar.

Comment (repher) [inline composer.json]: sorry ... my mistake. happend unintentially ... i'll correct that

**Comment by dereuromark** [comment]
The problem with this is that you cannot just overwrite the existing ones as "app ones".
Why would you not just rename the plugin one then?

**Comment by repher** [comment]
but why not? works great ...
```php
namespace App\Shell\Task;

use Cake\Core\Configure;
use Queue\Shell\Task\QueueEmailTask as ToolsEmailTask;

/**
 * A Simple QueueTask example.
 *
 */
class QueueEmailTask extends ToolsEmailTask
{

    public function run($data, $id = null) {
        $data['settings']['subject'] = Configure::read('Email.subject_pre') . ' ' . $data['settings']['subject'];
        parent::run($data, $id);
    }
}
```
so no code has to be changed anywhere because the task name stays the same ...

**Comment by dereuromark** [comment]
But what if you overwite one of the existing tasks in your APP directly?
That would now not work anymore.
As such the APP should come first, then all other plugins and last the Queue plugin.

**Comment by repher** [comment]
It does not work to overwrite a QueueTask within my APP without this change. As you see I wanted to extend the QueueEmailTask. As far as the code above goes, it works. But when Shell.php loads all tasks $tasks (which was filled in QueueShell::initialize()), it overwrites all APP tasks with QueueTasks with the same name ...
Added some debug output to Shell.php
```php
public function loadTasks()
{
    debug($this->tasks);
    if ($this->tasks === true || empty($this->tasks) || empty($this->Tasks)) {
        return true;
    }
    $this->_taskMap = $this->Tasks->normalizeArray((array)$this->tasks);
    debug($this->_taskMap);
    $this->taskNames = array_merge($this->taskNames, array_keys($this->_taskMap));
    return true;
}
```
leads to:
```
###########################
/src/Console/Shell.php (line 242)
########## DEBUG ##########
[
	(int) 0 => 'QueueEmail',
	(int) 1 => 'Queue.QueueEmail',
	(int) 2 => 'Queue.QueueExample',
	(int) 3 => 'Queue.QueueExecute',
	(int) 4 => 'Queue.QueueLongExample',
	(int) 5 => 'Queue.QueueRetryExample',
	(int) 6 => 'Queue.QueueSuperExample'
]
###########################
/src/Console/Shell.php (line 244)
########## DEBUG ##########
[
	'QueueEmail' => [
		'class' => 'Queue.QueueEmail',
		'config' => []
	],
	'QueueExample' => [
		'class' => 'Queue.QueueExample',
		'config' => []
	],
	'QueueExecute' => [
		'class' => 'Queue.QueueExecute',
		'config' => []
	],
	'QueueLongExample' => [
		'class' => 'Queue.QueueLongExample',
		'config' => []
	],
	'QueueRetryExample' => [
		'class' => 'Queue.QueueRetryExample',
		'config' => []
	],
	'QueueSuperExample' => [
		'class' => 'Queue.QueueSuperExample',
		'config' => []
	]
]
###########################
```
APP task gone ...

**Comment by repher** [comment]
This is because of [this](https://github.com/cakephp/cakephp/blob/28794510ef23b96049a22cde686e442a7e64ee9a/src/Core/ObjectRegistry.php#L242). Last name wins ...

**Comment by dereuromark** [comment]
So your fix fixes all that at once? Sounds good then.

**Comment by repher** [comment]
well I hope so :-)

**Comment by dereuromark** [comment]
Can you please confirm that? :) Then I can merge.

**Comment by repher** [comment]
It worked perfectly since this changes! Thanks!

**Pull request closed without merging**